src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,86 @@
+package seedu.address.model.task;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.TaskBuilder;
+
+public class TaskNameContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        TaskNameContainsKeywordsPredicate firstPredicate =
+                new TaskNameContainsKeywordsPredicate(firstPredicateKeywordList);
+        TaskNameContainsKeywordsPredicate secondPredicate =
+                new TaskNameContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        TaskNameContainsKeywordsPredicate firstPredicateCopy =
+                new TaskNameContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different task -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_taskNameContainsKeywords_returnsTrue() {
+        // One keyword
+        TaskNameContainsKeywordsPredicate predicate =
+                new TaskNameContainsKeywordsPredicate(Collections.singletonList("cs2103t"));
+        assertTrue(predicate.test(new TaskBuilder().withName("do cs2103t do cs2100").build()));
+
+        // Multiple keywords
+        predicate = new TaskNameContainsKeywordsPredicate(Arrays.asList("cs2103t", "cs2100"));
+        assertTrue(predicate.test(new TaskBuilder().withName("do cs2103t do cs2100").build()));
+
+        // Only one matching keyword
+        predicate = new TaskNameContainsKeywordsPredicate(Arrays.asList("cs2100", "cs2101"));
+        assertTrue(predicate.test(new TaskBuilder().withName("do cs2103t do cs2101").build()));
+
+        // Mixed-case keywords
+        predicate = new TaskNameContainsKeywordsPredicate(Arrays.asList("cS2103T", "Cs2100"));
```

Comment:
why some cs2103'T' capital, some small letters?
Isit intentional? HAHAHA sry I zoning out :(

Reply:
it's intentional - to test for mixed-case keywords i.e. "cS2103T" would match "cs2103t" as seen in TaskNameContainsKeywordsPredicate.java

```    @Override
    public boolean test(Task task) {
        return keywords.stream()
                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(task.getName().taskName, keyword));
    }
```